formats/formats.go

```diff
@@ -114,8 +115,8 @@ func IsHostname(s string) bool {
 			return false
 		}
 
-		// labels could not start with a digit or with a hyphen
-		if first := s[0]; (first >= '0' && first <= '9') || (first == '-') {
+		// labels must not start with a hyphen
```

Comment:
add 2nd line in comment:

//  RFC 1123 section 2.1: restriction on the first character 
//  is relaxed to allow either a letter or a digit